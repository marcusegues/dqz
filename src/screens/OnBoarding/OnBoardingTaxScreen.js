// @flow
import React from 'react';
import { translate } from 'react-i18next';
// $FlowFixMe
import { Text, View } from 'react-native';
import OnBoardingContainer from './subcomponents/OnBoardingContainer';
import OnBoardingParagraph from './subcomponents/OnBoardingParagraph';
import DoneButton from './subcomponents/DoneButton';
import { moderateScale } from '../../styles/Scaling';
import { MAIN_RED } from '../../styles/colors';
import type { TFunction } from '../../types/generalTypes';

const ownStyles = {
  container: {
    alignItems: 'center',
  },
  percents: {
    fontFamily: 'roboto_regular',
    fontSize: moderateScale(36),
    color: '#4a4a4a',
    paddingVertical: moderateScale(5),
  },
  bottomText: {
    fontFamily: 'roboto_medium',
    fontSize: moderateScale(14),
    color: MAIN_RED,
  },
};

type OnBoardingTaxScreenProps = {
  t: TFunction,
  navigation: any,
};

const OnBoardingTaxScreen = ({ navigation, t }: OnBoardingTaxScreenProps) => (
  <OnBoardingContainer>
    <View style={ownStyles.container}>
      <OnBoardingParagraph text={t('confirmationVat')} />
      <Text style={ownStyles.percents}>7.7%</Text>
      <OnBoardingParagraph text={t('confirmationVat2')} />
    </View>
    <Text style={ownStyles.bottomText}>
      {(t('vatInformation') || '').toUpperCase()}
    </Text>
    <DoneButton onPress={() => navigation.navigate('MainMenu')} />
  </OnBoardingContainer>
);

export default translate(['onBoarding'])(OnBoardingTaxScreen);
