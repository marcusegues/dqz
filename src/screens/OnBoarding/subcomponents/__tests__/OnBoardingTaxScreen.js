/* global describe, test, expect, jest */
import React from 'react';
import { shallow } from 'enzyme';

import OnBoardingTaxScreen from '../../OnBoardingTaxScreen';

jest.mock('react-i18next', () => ({
  translate: () => Component => props => <Component t={() => ''} {...props} />,
}));

describe('OnBoardingTaxScreen', () => {
  test('renders according to snapshot', () => {
    const navMock = jest.fn();
    const component = shallow(
      <OnBoardingTaxScreen navigation={{ navigate: navMock }} />
    ).dive();
    expect(component).toMatchSnapshot();
    component.find('DoneButton').forEach(child => {
      child.simulate('press');
    });
    expect(navMock).toBeCalled();
  });
});
