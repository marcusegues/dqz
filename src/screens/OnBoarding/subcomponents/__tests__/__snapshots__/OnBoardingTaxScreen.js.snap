// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnBoardingTaxScreen renders according to snapshot and DoneButton fires navigation 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OnBoardingTaxScreenInner
    navigation={
      Object {
        "dispatch": [Function],
      }
    }
    t={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <OnBoardingParagraph
            text=""
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#4a4a4a",
                "fontFamily": "roboto_regular",
                "fontSize": 56.57142857142857,
                "paddingVertical": 7.857142857142857,
              }
            }
          >
            7.7%
          </Text>
          <OnBoardingParagraph
            text=""
          />
        </View>,
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": 41,
              "width": 224,
            }
          }
        />,
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#757575",
                "fontFamily": "roboto_medium",
                "fontSize": 14,
              }
            }
          >
            
          </Text>
        </TouchableOpacity>,
        <DoneButton
          onPress={[Function]}
        />,
        <Unknown
          onPress={[Function]}
          text=""
        />,
        <Unknown
          modalVisible={false}
          navigation={
            Object {
              "dispatch": [Function],
            }
          }
          onConfirm={[Function]}
          onPressLegal={[Function]}
          toggleModalVisible={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <OnBoardingParagraph
              text=""
            />,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#4a4a4a",
                  "fontFamily": "roboto_regular",
                  "fontSize": 56.57142857142857,
                  "paddingVertical": 7.857142857142857,
                }
              }
            >
              7.7%
            </Text>,
            <OnBoardingParagraph
              text=""
            />,
          ],
          "style": Object {
            "alignItems": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "text": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "7.7%",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#4a4a4a",
                "fontFamily": "roboto_regular",
                "fontSize": 56.57142857142857,
                "paddingVertical": 7.857142857142857,
              },
            },
            "ref": null,
            "rendered": "7.7%",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "text": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "resizeMode": "contain",
          "source": 1,
          "style": Object {
            "height": 41,
            "width": 224,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#757575",
                "fontFamily": "roboto_medium",
                "fontSize": 14,
              }
            }
          >
            
          </Text>,
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#757575",
              "fontFamily": "roboto_medium",
              "fontSize": 14,
            },
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onPress": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onPress": [Function],
          "text": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "modalVisible": false,
          "navigation": Object {
            "dispatch": [Function],
          },
          "onConfirm": [Function],
          "onPressLegal": [Function],
          "toggleModalVisible": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <OnBoardingParagraph
              text=""
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#4a4a4a",
                  "fontFamily": "roboto_regular",
                  "fontSize": 56.57142857142857,
                  "paddingVertical": 7.857142857142857,
                }
              }
            >
              7.7%
            </Text>
            <OnBoardingParagraph
              text=""
            />
          </View>,
          <Image
            resizeMode="contain"
            source={1}
            style={
              Object {
                "height": 41,
                "width": 224,
              }
            }
          />,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#757575",
                  "fontFamily": "roboto_medium",
                  "fontSize": 14,
                }
              }
            >
              
            </Text>
          </TouchableOpacity>,
          <DoneButton
            onPress={[Function]}
          />,
          <Unknown
            onPress={[Function]}
            text=""
          />,
          <Unknown
            modalVisible={false}
            navigation={
              Object {
                "dispatch": [Function],
              }
            }
            onConfirm={[Function]}
            onPressLegal={[Function]}
            toggleModalVisible={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <OnBoardingParagraph
                text=""
              />,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#4a4a4a",
                    "fontFamily": "roboto_regular",
                    "fontSize": 56.57142857142857,
                    "paddingVertical": 7.857142857142857,
                  }
                }
              >
                7.7%
              </Text>,
              <OnBoardingParagraph
                text=""
              />,
            ],
            "style": Object {
              "alignItems": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "text": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "7.7%",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#4a4a4a",
                  "fontFamily": "roboto_regular",
                  "fontSize": 56.57142857142857,
                  "paddingVertical": 7.857142857142857,
                },
              },
              "ref": null,
              "rendered": "7.7%",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "text": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "resizeMode": "contain",
            "source": 1,
            "style": Object {
              "height": 41,
              "width": 224,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#757575",
                  "fontFamily": "roboto_medium",
                  "fontSize": 14,
                }
              }
            >
              
            </Text>,
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#757575",
                "fontFamily": "roboto_medium",
                "fontSize": 14,
              },
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onPress": [Function],
            "text": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "modalVisible": false,
            "navigation": Object {
              "dispatch": [Function],
            },
            "onConfirm": [Function],
            "onPressLegal": [Function],
            "toggleModalVisible": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
